Revisit TestNestedStackFails test

When switching from Travis to GitHub Actions, @EvanBoyle and I noticed that this test: https://github.com/pulumi/pulumi/blob/46da25492e38607967a83c0fa4bfa3f9772cb409/sdk/go/x/auto/local_workspace_test.go#L421

```
--- FAIL: TestNestedStackFails (12.24s)
    local_workspace_test.go:439: {Updating (pulumi/int_test12158168)

        View Live: https://app.pulumi.com/pulumi/parent/int_test12158168/updates/1

        + pulumi:pulumi:Stack: (create)
            [urn=urn:pulumi:int_test12158168::parent::pulumi:pulumi:Stack::parent-pulumi/parent/int_test12158168]
        Resources:
            + 1 created

        Policy Packs run:
            Name                      Version
            pulumi-internal-policies  6

        Duration: 7s
         engine: 127.0.0.1:57225resmon: 127.0.0.1:57230 map[] {update 2020-09-08T18:05:41.000Z  map[exec.kind:auto.inline] map[] succeeded 0xc000392630 0xc0003aa070}}
    local_workspace_test.go:441:
        	Error Trace:	local_workspace_test.go:441
        	Error:      	An error is expected but got nil.
        	Test:       	TestNestedStackFails
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x20658a2]

goroutine 918 [running]:
testing.tRunner.func1.1(0x2152b00, 0x399a980)
	/usr/local/opt/go/libexec/src/testing/testing.go:1057 +0x30d
testing.tRunner.func1(0xc000328900)
	/usr/local/opt/go/libexec/src/testing/testing.go:1060 +0x41a
panic(0x2152b00, 0x399a980)
	/usr/local/opt/go/libexec/src/runtime/panic.go:969 +0x175
github.com/pulumi/pulumi/sdk/v2/go/x/auto.TestNestedStackFails(0xc000328900)
	/Users/lbriggs/src/git/pulumi/pulumi/sdk/go/x/auto/local_workspace_test.go:442 +0x882
testing.tRunner(0xc000328900, 0x242a688)
	/usr/local/opt/go/libexec/src/testing/testing.go:1108 +0xef
created by testing.(*T).Run
	/usr/local/opt/go/libexec/src/testing/testing.go:1159 +0x386
FAIL	github.com/pulumi/pulumi/sdk/v2/go/x/auto	179.643s
FAIL
```

Started to fail. After some brief debugging work, it seems that we need to check the output of this https://github.com/pulumi/pulumi/blob/46da25492e38607967a83c0fa4bfa3f9772cb409/sdk/go/x/auto/stack.go#L790

to ensure the content is as expected

--- a/sdk/go/auto/local_workspace_test.go
+++ b/sdk/go/auto/local_workspace_test.go
@@ -1045,8 +1045,6 @@ func TestUpsertStackInlineSourceParallel(t *testing.T) {
 func TestNestedStackFails(t *testing.T) {
 	t.Parallel()
 
-	// FIXME: see https://github.com/pulumi/pulumi/issues/5301
-	t.Skip("skipping test, see pulumi/pulumi#5301")
 	testCtx := context.Background()
 	sName := randomStackName()
 	parentstackName := FullyQualifiedStackName(pulumiOrg, "parent", sName)

